src/BenchmarkDotNet/Validators/ParamsValidator.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+
+namespace BenchmarkDotNet.Validators
+{
+    public class ParamsValidator : IValidator
+    {
+        public static readonly ParamsValidator FailOnError = new ();
+
+        public bool TreatsWarningsAsErrors => true;
+
+        public IEnumerable<ValidationError> Validate(ValidationParameters input) => input.Benchmarks
+            .Select(benchmark => benchmark.Descriptor.Type)
+            .Distinct()
+            .SelectMany(Validate);
+
+        private IEnumerable<ValidationError> Validate(Type type)
+        {
+            var constantsWithParams = type
+                .GetFields(BindingFlags.Public | BindingFlags.Static | BindingFlags.FlattenHierarchy)
+                .Where(fieldInfo => fieldInfo.IsLiteral && !fieldInfo.IsInitOnly && fieldInfo.HasAttribute<ParamsAttribute>());
```

Comment:
```dif
- fieldInfo.IsLiteral && !fieldInfo.IsInitOnly
+ fieldInfo.IsLiteral || fieldInfo.IsInitOnly
```

We should check for both const and readonly.

Reply:
Also should check for `ParamsAllValuesAttribute` and `ParamsSourceAttribute` (unless you want to duplicate this check in the `ParamsAllValuesValidator` and add a `ParamsAllValuesValidator`).